app/src/main/java/org/mifos/mobile/ui/beneficiary_list/BeneficiaryListViewModel.kt

```diff
@@ -30,4 +45,19 @@ class BeneficiaryListViewModel @Inject constructor(private val beneficiaryReposi
             }
         }
     }
-}
\ No newline at end of file
+}
+
+sealed class BeneficiaryUiState {
+    object Initial : BeneficiaryUiState()
+    object Loading : BeneficiaryUiState()
+    object CreatedSuccessfully : BeneficiaryUiState()
+    object UpdatedSuccessfully : BeneficiaryUiState()
+    object DeletedSuccessfully : BeneficiaryUiState()
+    data class ShowError(val message: Int) : BeneficiaryUiState()
+    data class SetVisibility(val visibility: Int) : BeneficiaryUiState()
+    data class ShowBeneficiaryTemplate(val beneficiaryTemplate: BeneficiaryTemplate) :
+        BeneficiaryUiState()
+
+    data class ShowBeneficiaryList(val beneficiaries: List<Beneficiary?>) : BeneficiaryUiState()
+
+}
```

Comment:
Every screen should have its dedicated UIState, show make separate `BeneficiaryListUiState`  with only its dedicated states. It should have only Loading, Error, and Success. Initial should be in case when we are showing content irrespective of uiStates